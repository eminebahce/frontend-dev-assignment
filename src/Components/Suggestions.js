import React from 'react';

export default function Suggestions (props) {
    return(
        <ul data-test="suggestions">
            {props.suggestionsList && props.suggestionsList.map((item, i) => <li key={i}>{item.searchterm} {item.nrResults}</li>)}
        </ul>
    )
};