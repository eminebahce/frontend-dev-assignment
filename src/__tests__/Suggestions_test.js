import React from 'react';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
import Suggestions from '../Components/Suggestions';
import renderer from 'react-test-renderer';

describe('<Suggestions /> component', () => {
   it('should render without error', () => {
       const component = shallow(<Suggestions />);
       expect(component.find(`[data-test='suggestions']`)).toBeTruthy();
   }) ;

   describe('renders sugestions list', () => {
      it('should render without error', () => {
          const component = renderer.create(
              <Suggestions />
          );
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
      });
   });
});