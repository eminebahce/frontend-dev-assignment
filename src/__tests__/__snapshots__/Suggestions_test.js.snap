// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Suggestions /> component renders sugestions list should render without error 1`] = `
<ul
  data-test="suggestions"
/>
`;
